src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -25,6 +26,7 @@ const usePushNotificationHook = () => {
   const {localDb} = useLocalDatabaseHook();
   const {fetchChannelDetail, setSelectedChannel} = useChatUtilsHook();
   const {updateAppBadgeFromDB} = useAppBadgeHook();
+  const {setIsLoadingFetchAllMessage} = useChatScreenHook();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The usage of `setIsLoadingFetchAllMessage` from `useChatScreenHook` here might be problematic. Hooks should ideally be used at the top level of your React function components. Using hooks inside callbacks, loops, conditions, or nested functions can potentially lead to bugs and inconsistencies in the UI.

```diff
+ const {setIsLoadingFetchAllMessage} = useChatScreenHook();
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->